src/main/java/org/opensearch/agent/tools/RAGTool.java

```diff
@@ -264,32 +233,20 @@ public RAGTool create(Map<String, Object> params) {
                         .builder()
                         .client(client)
                         .xContentRegistry(xContentRegistry)
-                        .index(index)
-                        .embeddingField(embeddingField)
-                        .sourceFields(sourceFields)
-                        .embeddingModelId(embeddingModelId)
-                        .k(k)
-                        .docSize(docSize)
                         .inferenceModelId(inferenceModelId)
-                        .queryType(queryType)
+                        .enableContentGeneration(enableContentGeneration)
                         .queryTool(neuralSparseSearchTool)
                         .build();
                 case "neural":
                     VectorDBTool.Factory.getInstance().init(client, xContentRegistry);
-                    params.put(VectorDBTool.MODEL_ID_FIELD, EMBEDDING_MODEL_ID_FIELD);
+                    params.put(VectorDBTool.MODEL_ID_FIELD, embeddingModelId);
```

Comment:
We should also do this for NeuralSparseTool

Reply:
We don't need to call `VectorDBTool.Factory.getInstance().init(client, xContentRegistry)` or `NeuralSparseSearchTool.Factory.getInstance().init(client, xContentRegistry)` here. This has been done during the initialization of skills plugin

Reply:
added the initiation to the test to simulate the initiations of tools in  [ToolPlugins.java](https://github.com/mingshl/skills/blob/5d7b96281e9c218416c9f85c60e6904bc074c1d7/src/main/java/org/opensearch/agent/ToolPlugin.java#L68) 

mapped the inference_model_id to model_id for NeuralSparseTool